Patch-Source: https://gitlab.com/mbunkus/mkvtoolnix/-/commit/b317f8dd6ae4b6a1ac70e56c458063fab7a4e991.patch
--
From b317f8dd6ae4b6a1ac70e56c458063fab7a4e991 Mon Sep 17 00:00:00 2001
From: Moritz Bunkus <mo@bunkus.online>
Date: Sun, 20 Aug 2023 18:28:03 +0200
Subject: [PATCH] DTS reader: fix linker error wrt. static class member vars

On some compiler versions with some options the previous way seemed to
cause errors wrt. undefined references. So give the variable a fixed
place to live.
---
 src/input/r_dts.cpp | 2 ++
 src/input/r_dts.h   | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/input/r_dts.cpp b/src/input/r_dts.cpp
index 5094d9e57..692a9a777 100644
--- a/src/input/r_dts.cpp
+++ b/src/input/r_dts.cpp
@@ -25,6 +25,8 @@
 #include "merge/input_x.h"
 #include "output/p_dts.h"
 
+unsigned int const dts_reader_c::s_buf_size = 128 * 1000;
+
 bool
 dts_reader_c::probe_file() {
   m_chunks        = scan_chunks(*m_in);
diff --git a/src/input/r_dts.h b/src/input/r_dts.h
index 69e990062..7db299ad8 100644
--- a/src/input/r_dts.h
+++ b/src/input/r_dts.h
@@ -55,7 +55,7 @@ private:
   chunks_t m_chunks;
   chunks_t::const_iterator m_current_chunk;
 
-  static auto const s_buf_size = 128 * 1000;
+  static unsigned int const s_buf_size;
 
 public:
   dts_reader_c();
-- 
GitLab

